### ISSUE TITLE
Issue Building client-sdk-js: Rollup Errors and Incomplete Bindings

### ISSUE THREAD

**Issue by CSantosM** [livekit/client-sdk-js#1015]
### Describe the bug

Hi there,

I'm currently facing difficulties while attempting to build the client-sdk-js, encountering issues with both the **build:watch** and **build** scripts.

**build:watch** Script:
After installing the dependencies and executing the build:watch script, Rollup throws an error, preventing successful execution.

**build** Script:
Running the build script results in warnings, and when I attempt to install the generated build in a project, there seem to be missing bindings for certain methods. This leads me to believe that the build process might not be completing properly.

I would appreciate any guidance or assistance in resolving these issues. If there are specific steps or considerations I might be missing during the build process, please let me know.

Thanks in advance for your help!

![image](https://github.com/livekit/client-sdk-js/assets/10628395/29adcbf9-c20d-4fcd-9a0a-077f739051ef)

### Reproduction

```
pnpm install 
pnpm build:watch
````

### System Info

```shell
Ubuntu
Node v20.11.0
Building client-sdk-js from master
```

### Severity

blocking all usage of LiveKit

**Comment by lukasIO** [comment]
thanks for the report, try by adding `--bundleConfigAsCjs` to the command in the `build:watch` script. 
Will open a PR to address this. 
Note that it might require two watch processes to run though, as the worker for E2EE is built with a separate rollup config

**Comment by CSantosM** [comment]
After adding the flag mentioned in the build:watch script, the following warnings appear. If unfortunately it is part of the third party element (rollup) and nothing can be done, we can ignore it. 

![image](https://github.com/livekit/client-sdk-js/assets/10628395/8f1907e1-1632-4f0b-b9b3-b1f102ee4f32)

I have verified that with the introduced flag, the build:watch script now runs without any issues.

Added PR here https://github.com/livekit/client-sdk-js/pull/1017
